DataFormats/CTPPSReco/interface/CTPPSLocalTrackLite.h

```diff
@@ -11,75 +11,150 @@
+        unsigned short pNumberOfPointsUsedForFit, float pt, float ptu)
+      : rpId(pid), x(px), x_unc(pxu), y(py), y_unc(pyu), tx(ptx), tx_unc(ptxu), ty(pty), ty_unc(ptyu), chiSquaredOverNDF(pchiSquaredOverNDF), 
+      pixelTrack_reco_info(ppixelTrack_reco_info), numberOfPointsUsedForFit(pNumberOfPointsUsedForFit), time(pt), time_unc(ptu)
+    { 
     }
 
     /// returns the RP id
-    uint32_t getRPId() const
+    inline uint32_t getRPId() const
     {
       return rpId;
     }
 
     /// returns the horizontal track position
-    float getX() const
+    inline float getX() const
     {
       return x;
     }
 
     /// returns the horizontal track position uncertainty
-    float getXUnc() const
+    inline float getXUnc() const
     {
       return x_unc;
     }
 
     /// returns the vertical track position
-    float getY() const
+    inline float getY() const
     {
       return y;
     }
 
     /// returns the vertical track position uncertainty
-    float getYUnc() const
+    inline float getYUnc() const
     {
       return y_unc;
     }
 
     /// returns the track time
-    float getTime() const
+    inline float getTime() const
     {
       return time;
     }
 
     /// returns the track time uncertainty
-    float getTimeUnc() const
+    inline float getTimeUnc() const
     {
       return time_unc;
     }
 
+    /// returns the track horizontal angle
+    inline float getTx() const
+    {
+        return tx;
+    }
+
+    /// returns the track horizontal angle uncertainty
+    inline float getTxUnc() const
+    {
+        return tx_unc;
+    }
+    
+    /// returns the track vertical angle
+    inline float getTy() const
+    {
+        return ty;
+    }
+    
+    /// returns the track vertical angle uncertainty
+    inline float getTyUnc() const
+    {
+        return ty_unc;
+    }
+    
+    /// returns the track fit chi Squared over NDF
+    inline float getChiSquaredOverNDF() const
+    {
+        return chiSquaredOverNDF;
+    }
+
+    /// returns the track reconstruction info byte
+    inline CTPPSReconstructionInfo getPixelTrackRecoInfo() const
+    {
+        return pixelTrack_reco_info;
+    }
+
+    /// returns the number of points used for fit
+    inline unsigned short getNumberOfPointsUsedForFit() const
+    {
+        return numberOfPointsUsedForFit;
+    }   
+
   protected:
     /// RP id
     uint32_t rpId;
 
+    /// local track parameterization
+    /// x = x0 + tx*(z-z0), y = y0 + ty*(z-z0)
+    /// x0, y0, z-z0 in mm
+    /// z0: position of the reference scoring plane (in the middle of the RP)
+
     /// horizontal hit position and uncertainty, mm
     float x, x_unc;
 
     /// vertical hit position and uncertainty, mm
     float y, y_unc;
 
+    /// horizontal angle and uncertainty, x = x0 + tx*(z-z0)
+    float tx, tx_unc;
+
+    /// vertical angle and uncertainty, y = y0 + ty*(z-z0)
+    float ty, ty_unc;
+
+    /// fit chi^2 over NDF
+    float chiSquaredOverNDF;
```

Comment:
(a comment to all float data members)

from the runtime checks it looks like the size of CTPPSLocalTrackLite about doubles.
It's worth to check if reduction is possible.
Please check if all the variables need the full float precision.
One easy option is  rounding the mantissa with e.g. reduceMantissaToNbitsRounding from DataFormats/Math/interface/libminifloat.h (in miniAOD we commonly round to between 8 and 12 bits)

Reply:
@AndreaBellora 
please clarify if this proposal for rounding is something that you can apply.
Thank you.

Reply:
There has been a bit of "internal discussion" going on and it should be feasible. Could you please explain to me how the reduction works? I see that the reduceMantissaToNbitsRounding function is usually applied to a float, but the rounded variable is again saved as a float. How does this reduce the size of the variable? Is this something that happens when data is compressed into root files? 
Thank you

Reply:
right, it's after compression.
There is no concern here about the memory use during run time.

Reply:
That's perfect, I'm double-checking the precision requirements and adding the code. Thanks a lot.